contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/common/implementation/ExpandedERC20.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleInterface.sol";
+
+// TODO use OptimisticOracleInterface from @uma/core once it's updated with setEventBased
+interface OptimisticOracleInterfaceEventBased {
+    function setEventBased(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData
+    ) external;
+}
+
+contract EventBasedPredictionMarket is Testable {
+    using SafeERC20 for ExpandedERC20;
+
+    /***************************************************
+     *  EVENT BASED PREDICTION MARKET DATA STRUCTURES  *
+     ***************************************************/
+    bool public priceRequested;
+    bool public receivedSettlementPrice;
+
+    uint256 public expirationTimestamp;
+    string public pairName;
+    uint256 public collateralPerPair = 1e18; // Amount of collateral a pair of tokens is always redeemable for.
```

Comment:
Couldn't we just remove `collateralPerPair` and any accompanying logic? If `expiryPercentLong` is always 0-1 and `collateralPerPair` always defaults to `1` (this was a good choice imo), then we don't ever need to translate between a settlement price and what it means in backing collateral. So we could just use the direct settlement price in place of things like: https://github.com/UMAprotocol/dev-quickstart/pull/1/files#diff-f8cf1cf787f10b721ee1e17c6d09988c8f6efa9a0e0b79a5f41cb9dcf4b86925R218

Potentially should also rename expiryPercentLong to something like `settlementPrice`, since there isn't really a concept of an expiryPercentLong since this is always equivalent to what the returned settlement price would be.